packages/@lwc/synthetic-shadow/src/3rdparty/polymer/path-composer.ts

```diff
@@ -18,22 +14,29 @@ The complete set of contributors may be found at http://polymer.github.io/CONTRI
 Code distributed by Google as part of the polymer project is also
 subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
 */
+
+import { getOwnerDocument } from '../../shared/utils';
+import { isNull } from '../../shared/language';
+
+const GlobalShadowRoot = (window as any).ShadowRoot;
+
 export function pathComposer(startNode: EventTarget, composed: boolean): EventTarget[] {
```

Comment:
this needs detailed review @ravijayaramappa

Reply:
Seems very similar to the original implementation if you combine the first two `if`s.